File: JobPortal/src/main/java/com/jobportal/DTO/JobRequest.java
package com.jobportal.DTO;

import com.jobportal.Entities.EmploymentType;
import com.jobportal.Entities.JobCategory;
import com.jobportal.Entities.WorkMode;
import lombok.*;

import java.time.LocalDateTime;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class JobRequest {

    private String jobTitle;

    private String jobDescription;

    private String jobLocation;

    private EmploymentType employmentType;

    private JobCategory category;

    private WorkMode workMode;

    private Integer experienceRequired;

    private Double minSalary;

    private Double maxSalary;

    private Integer vacancies;

    private String benefits;

    private Boolean active;

    private LocalDateTime expiryDate;

    private Set<Long> skillIds;
}

File: JobPortal_UI/src/app/interface/job-request.ts
export interface JobDTO {
  jobId: number;
  companyName?: string;
  skillNames?: string[];
  createdAt?: string;

  jobTitle: string;
  jobDescription: string;
  jobLocation: string;

  employmentType: string;
  category: string;
  workMode: string;

  experienceRequired: number;
  minSalary: number;
  maxSalary: number;
  vacancies: number;

  benefits?: string;
  active?: boolean;
  expiryDate: string;

  skillIds?: number[];
}